object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -445,107 +343,100 @@ private Node removeCasts(Node node) {
   }
 
   private void handleAssignFromRHS(
-      AssignmentNode node, Set<ImmutableSet<LocalVarWithTree>> newDefs, Node rhs) {
+      AssignmentNode node, Set<ImmutableSet<LocalVarWithTree>> newDefs, Node rhstemp) {
     Node lhs = node.getTarget();
     Element lhsElement = TreeUtils.elementFromTree(lhs.getTree());
+    if (typeFactory.mapTempVarToNode.inverse().containsKey(rhstemp.getTree())) {
```

Comment:
This logic is weird.  Why did we rename the parameter to `rhstemp`, which is kind of strange?  Can we write this code so it doesn't re-assign the parameter?